mypy/semanal.py

```diff
@@ -2743,7 +2747,11 @@ def visit_yield_expr(self, expr: YieldExpr) -> None:
             self.fail("'yield' outside function", expr, True, blocker=True)
         else:
             if self.function_stack[-1].is_coroutine:
-                self.fail("'yield' in async function", expr, True, blocker=True)
+                if self.options.python_version < (3, 6):
+                    self.fail("'yield' in async function", expr, True, blocker=True)
+                else:
+                    self.function_stack[-1].is_generator = True
+                    self.function_stack[-1].is_async_generator = True
```

Comment:
This is not important, but it looks a bit strange that you need to set _both_ flags here.